src/parser/mod.rs

```diff
@@ -257,6 +259,7 @@ pub struct Parser<'a> {
     options: ParserOptions,
     /// ensure the stack does not overflow by limiting recursion depth
     recursion_counter: RecursionCounter,
+    max_depth: usize,
```

Comment:
Please document what this field is foe.

Also, could you please explain why you didn't use the counter in `RecusionCounter`?